interface/src/app/node/node.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * "adage.gene.network" module.
+ */
+
+angular.module('adage.node', [
+  'ui.router',
+  'placeholders',
+  'ui.bootstrap'
+])
+
+.config(function config($stateProvider) {
+  $stateProvider.state('node', {
+    url: '/node/{id:int}',
+    views: {
+      main: {
+        templateUrl: 'node/node.tpl.html',
+        controller: 'NodeCtrl as ctrl'
+      }
+    },
+    data: {pageTitle: 'Node Information'}
+  });
+})
+
+.factory('NodeInfo', ['$resource', '$stateParams',
+  function($resource) {
+    return $resource('/api/v0/node/:id');
+  }
+])
+
+.factory('HeavyGenes', ['$resource', function($resource) {
+  return $resource('/api/v0/participation/');
+}])
+
+.controller('NodeCtrl', ['NodeInfo', '$stateParams', '$log',
+  function NodeController(NodeInfo, $stateParams, $log) {
+    var self = this;
+    if (!$stateParams.id) {
+      self.statusMessage = 'Please specify node ID in the URL.';
+      return;
+    }
+    self.id = $stateParams.id;
+    self.statusMessage = 'Connecting to the server ...';
+    NodeInfo.get(
+      {id: self.id},
+      function success(response) {
+        self.name = response.name;
+        self.mlmodel = response.mlmodel.title;
+        self.statusMessage = '';
+      },
+      function error(err) {
+        $log.error('Failed to get node information: ');
+        self.statusMessage = 'Failed to get node information from server';
+      }
+    );
+  }
+])
+
+.directive('highWeightGenes', ['HeavyGenes', '$log',
```

Comment:
Yes, when AngularJS executes link function (behind the scene), it calls the function in the following fashion:
```
link($scope, DOM_element, attributes_of_DOM_element, current_controller, transclude)
```
That is why sometimes link is defined like this:
```
link: function($scope, $element, $attrs, ctrl, $transclude)
```
(But this kind of usage seems rare.)